### PR TITLE
Unpin NumPy, Pandas, ASV, Support Python 3.12, Fix Broken Install Link 

### LINKED ISSUE #885: Add Python 3.12 to CI

### PR DESCRIPTION
**Pull request by philipc2**
Closes #885, #882, #884

## Overview

* Unpins `numpy`, `pandas` and `asv` packages
* Adds Python 3.12 to CI workflows
* Fixes broken install link in README

### CODE REVIEW

**Review by rajeeja — DISMISSED** [review]

**Comment by philipc2** [comment]
@rajeeja 

Looks like switching to Numpy 2.0 didn't impact performance much. There is however a big improvement with unpinning Pandas for our GeoDataFrame construction.

**Comment by philipc2** [comment]
Once https://github.com/nschloe/pyfma/pull/17 is merged and released, the errors with `pyfma` should be fixed.

**Comment by philipc2** [comment]
@rajeeja 

Should be good for another review.

File: test/test_helpers.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def test_replace_fill_values_invalid(self):
         """Tests _replace_fill_values() helper function attempting to use a
         fill value that is not representable by the current dtype."""
 
-        face_nodes = np.array([[1, 2, -1], [-1, -1, -1]], dtype=np.uint32)
+        face_nodes = np.array([[1, 2, -1], [-1, -1, -1]], dtype=np.int32)
```
Comment (rajeeja) [inline test/test_helpers.py:189]: was this causing issues with `uint32`?

**Review by rajeeja — APPROVED** [review]
thanks for removing pins - so important to get this done.
Also closing issue #820

File: test/test_helpers.py (hunk 1 above)
Comment (philipc2) [reply test/test_helpers.py:189]: Yeah, which makes sense (`-1` can't be represented in `uint32`). Looks like NumPy 2.0 correctly handles this now.

**Pull request merged**